runtime/doc/repeat.txt

```diff
@@ -762,6 +765,35 @@ the command after changing the plugin help: >
 	:helptags path/start/foobar/doc
 	:helptags path/opt/fooextra/doc
 
+The messages that are in the lang/<lang_id>/LC_MESSAGES/foo.po file need to be
+translated to a format that the |gettext()| function understands by running the
+*msgfmt* program. This will result in a lang/<lang_id>/LC_MESSAGES/foo.mo
+file.
+
+In your plugin, you need to call the |bindtextdomain()| function as follows.
+This assumes that the directory structure is as above: >
+	:call bindtextdomain("foo", getcwd()~"../lang")
+<
+You only need to do this once. After this call, you can use: >
+	:echo gettext("Hello", "foo")
+>
+to get the text "Hello" translated to the user's preferred language (if the
+plugin messages have been translated to this language).
+
+If you need to use the message in a different encoding from the current one,
+you can use (for instance) >
+	:echo gettext("Hello", "foo", "utf-8")
+<or >
+	:echo gettext("Hello", "foo", "cp1251")
+
+To create the foo.po file, you need to create a foo.pot file first. The
+entries in this file need to be translated to the language(s) you want to be
+supported by your plugin.
+
+To create the foo.pot file, run the following command: >
+	cd ~/.vim/pack/start/foobar
+	make -f ~/src/vim/src/po/Makefile PACKAGE=foo PO_BASEDIR=~/src/vim/src/po PO_INPUTLIST= PO_VIM_JSLIST="plugin__foo.js plugin__bar.js autoload__foo.js" PO_VIM_INPUTLIST="plugin/foo.vim plugin/bar.vim autoload/foo.vim" foo.pot
+<
```

Comment:
that sounds difficult for a user, that just wants to translate some messages.

Reply:
It's not the translation per se, it's the plugin writer that wants to have their messages translated to another language... and for that, the Vim script needs to be 'converted' to "Javascript" in order to be read by the xgettext program. And I wasn't really happy with the way to run the Makefile, but I couldn't think of another way to get the plumbing to work...